shark/examples/shark_inference/stable_diffusion/utils.py

```diff
@@ -70,12 +72,40 @@ def compile_through_fx(
     model_name,
     is_f16=False,
     f16_input_mask=None,
+    use_tuned=False,
     extra_args=[],
 ):
 
     mlir_module, func_name = import_with_fx(
         model, inputs, is_f16, f16_input_mask
     )
+
+    if use_tuned:
+        model_name = model_name + "_tuned"
+        tuned_model_path = f"{args.annotation_output}/{model_name}_torch.mlir"
+        if not os.path.exists(tuned_model_path):
+            if "vae" in model_name.split("_")[0]:
+                args.annotation_model = "vae"
+
+            if "cuda" in args.device:
+                output_path = (
+                    f"{args.annotation_output}/{model_name}_orig.mlir"
+                )
+                with open(output_path, "w") as f:
+                    f.write(mlir_module)
+                    f.close()
+                mlir_module = output_path
+
+            tuned_model, tuned_model_path = sd_model_annotation(
```

Comment:
Don't you think we will have to check for `batch_size` `height` and `width` and `max_len` too when applying annotation?

Reply:
Good point! I only tested the codes with the default settings on vulkan. And the tuned configs only apply to batch_size 1, height/width 512. The max length could be 77 or 64 though. The goal of this PR is making the basic functionality working for the default settings. I will test and add more settings in the following patches.

Reply:
Cool! We can do `sys.exit("Tuned models are currently supported for ...")` if the above criteria aren't met.

Reply:
I think it should fall into untuned models if the tuned models are not available, similar to we used to check for device.